src/aggregator/aggregator/handler/filter/percentageFilter.go

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) 2023 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package filter
+
+import (
+	"math/rand"
+	"time"
+
+	"github.com/m3db/m3/src/msg/producer"
+)
+
+type percentageFilter struct {
+	percentage float64
+}
+
+// NewPercentageFilter creates a filter based on percentage.
+func NewPercentageFilter(percentage float64) producer.FilterFunc {
+	rand.Seed(time.Now().UnixNano())
+	f := percentageFilter{percentage: percentage}
+	return f.Filter
+}
+
+func (f percentageFilter) Filter(_ producer.Message) bool {
+	return f.percentage > rand.Float64()
```

Comment:
i don't know if you care about sampling the same ids or not - this isn't giving you that

Reply:
thanks for pointing that out, in this percentage filter I don't care.
if we want to sample the same ids, that will require id to be decoded from the producer.Message, that would be a totally difference type of filter